lib/workers/global/config/parse/env.ts

```diff
@@ -97,6 +101,30 @@ export function getConfig(inputEnv: NodeJS.ProcessEnv): AllConfig {
     try {
       config = JSON5.parse(env.RENOVATE_CONFIG);
       logger.debug({ config }, 'Detected config in env RENOVATE_CONFIG');
+
+      const { isMigrated, migratedConfig } = migrateConfig(config);
+      if (isMigrated) {
+        logger.warn(
+          { originalConfig: config, migratedConfig },
+          'RENOVATE_CONFIG needs migrating',
+        );
+        config = migratedConfig;
+      }
+
+      const { warnings, errors } = await validateConfig(true, migratedConfig);
+
+      if (warnings.length) {
+        logger.warn(
+          { warnings },
+          'Config validation warnings found in RENOVATE_CONFIG',
+        );
+      }
+      if (errors.length) {
+        logger.warn(
+          { errors },
+          'Config validation errors found in RENOVATE_CONFIG',
+        );
+      }
```

Comment:
Extract to shared function (`lib/workers/global/config/parse/util.ts`), same code is used on file.ts